packages/components/constraints/src/horizontal/horizontal.tsx

```diff
@@ -1,11 +1,40 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import { css } from '@emotion/react';
 import { filterDataAttributes } from '@commercetools-uikit/utils';
 import invariant from 'tiny-invariant';
 import { getMaxPropTokenValue, getMaxPropEquivalent } from '../helpers';
 
-function getConstraintStyles(maxProp, constraintProp) {
+type TMaxProp =
+  | 1
+  | 2
+  | 3
+  | 4
+  | 5
+  | 6
+  | 7
+  | 8
+  | 9
+  | 10
+  | 11
+  | 12
+  | 13
+  | 14
+  | 15
+  | 16
+  | 'scale'
+  | 'auto';
+
+type TConstraintProp = 'xs' | 's' | 'm' | 'l' | 'xl' | 'scale';
```

Comment:
To be deprecated and remove in the next major release. Keeping this for now to ease the friction.
If we remove this now, it becomes a breaking change and I'm not sure for the moment if I want to bake this into the typescript migration.